test/common/packagelib.py

```diff
@@ -138,8 +138,9 @@ def setUp(self):
     # Helper functions for creating packages/repository
     #
 
-    def createPackage(self, name, version, release, install=False,
-                      postinst=None, depends="", content=None, arch=None, provides=None, **updateinfo):
+    def createPackage(self, name: str, version: str, release: str, *, install: bool = False,
+                      postinst: str | None = None, depends: str = "", content: dict[str, dict[str, str] | str] | None = None,
```

Comment:
please use abstract types as appropriate (ie: `collections.abc.Mapping` instead of `dict`).

Reply:
hmm what is the difference here?

Reply:
A dict must be definitely a dict.  A mapping can be many things, most likely a dict.  In particular, Mapping only supports reading, so in any case where the intent is to pass a dict-like structure which isn't modified by the receiving function, Mapping is probably a better choice.

See also `Sequence` instead of `list`, `Collection` and `Container` instead of `set`, etc.

Reply:
Ack:

```
d = {}
>>> isinstance(d, Mapping)
```

Works.